jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -1514,6 +1514,62 @@ ecma_string_get_size (const ecma_string_t *string_p) /**< ecma-string */
 } /* ecma_string_get_size */
 
 /**
+ * Get size of UTF-8 encoded string size from ecma-string
+ *
+ * @return number of bytes in the buffer needed to represent an UTF-8 encoded string
+ */
+lit_utf8_size_t
+ecma_string_get_utf8_size (const ecma_string_t *string_p) /**< ecma-string */
+{
+  switch (ECMA_STRING_GET_CONTAINER (string_p))
+  {
+    case ECMA_STRING_CONTAINER_HEAP_UTF8_STRING:
+    {
+      if (ECMA_STRING_IS_ASCII ((const lit_utf8_byte_t *) (string_p + 1),
+                                (lit_utf8_size_t) string_p->u.utf8_string.size))
```

Comment:
Perhaps we could simply compare (lit_utf8_size_t) string_p->u.utf8_string.size with (lit_utf8_size_t) string_p->u.utf8_string.length. I feel calling a function is unnecessary because the check is very simple.